src/test/java/com/alem/todolist/ToDoListControllerTest.java

```diff
@@ -32,47 +39,70 @@
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockito.Mockito.*;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @RunWith(SpringRunner.class)
 @WebMvcTest(ToDoListController.class)
+@AutoConfigureMockMvc
 public class ToDoListControllerTest {
 
     @MockBean
-    ToDoListRepository toDoListRepository;
-
+    private ToDoListRepository toDoListRepository;
     @InjectMocks
-    ToDoListService toDoListService = new ToDoListServiceImpl(toDoListRepository);
+    private ToDoListController toDoListController = new ToDoListController(toDoListRepository);
 
     @Autowired
     private MockMvc mockmvc;
+
+    //  // task: id, desc, date, location, group
     @BeforeEach
     void setup() {
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
+        List<Task> tasks = toDoListController.getTasks();
+        when(toDoListRepository.findAll()).thenReturn(tasks);
     }
     @Test
     void shouldCreateMockMvc(){
         assertNotNull(mockmvc);
     }
 
-
     @Test
     void shouldReturnFirstTask() throws Exception {
-        this.mockmvc.perform(get("/tasks")
+        ResultActions result = this.mockmvc.perform(get("/tasks")
                         .param("id", "1")
                         .accept("application/json"))
                         .andExpect(status().isOk());
+        String content = result.andReturn().getResponse().getContentAsString();
```

Comment:
You should mock the database inserts and retrievals. So whenever the service layer calls the repository, you should have the proper mocks prepared.